Minor corrections

Maxime5G opened the pull request:
Hi Matthew,

Finally managed to push a fresh PR directly on the main repo. 

Here are some suggested changes for minor corrections (unused commas and imports) + I think what could fix the pipe() issues. There were several locations where the number of threads differed between rules (either workflow.cores or pipeline_config['threads']). I propose to standardize the number of threads across the snakefile using workflow.cores.

mattjmeier reviewed (changes requested):
Thanks so much Maxime for this contribution!

I appreciate the cleanup you've done here.

One minor thing - I think I see why it doesn't pass the workflow tests, it's actually because of the workflow.cores change. We do indeed want to control the number of threads separately for some processes - the reason is that for parallel processing, some tasks will plateau at a different number of CPUs than other tasks. This is why we have the pipeline config number of threads setting to be separate from the workflow.cores. The number of cores would be, for instance, if you had 32 CPUs available on your cluster, set to 32; but, you could run 4 processes of STAR simultaneously at 8 threads each, for example. This would actually run faster than doing a single STAR process one at a time, because you get no additional gains past a certain point.

It's a bit of a process to optimize these things for a given cluster size, but those examples I gave should give you a starting point.

The pipe() command failing was also related to the number of cores vs. threads, I think. If you change the num_threads back, and push that commit to this PR, it should rerun the tests and hopefully pass.

Also, did you rebase your branch against main? I think you might have done it before a recent change: https://github.com/R-ODAF/R-ODAF_Health_Canada/commit/0a2e28e445bc81c871f6ce7a0257794072251a60 , because it looks like you are re-committing this change.

For further information on the thread vs core optimization problem, see here:
- specifically some discussion as it relates to alignment: https://www.biostars.org/p/253707/
- general information on optimization: https://en.wikipedia.org/wiki/Amdahl%27s_law
- some snakemake documentation on how it allocates resources: https://snakemake.readthedocs.io/en/stable/tutorial/advanced.html#step-1-specifying-the-number-of-used-threads

Thank you again for your help with the project! Looking forward to working together on it more.

Maxime5G commented:
Hi Matthew,

I reverted some of the changes on the number of threads. I also modified the number of threads for the rule STAR_make_index as I still had resource allocation issues with workflow.cores. Now it seems to run on my side.

I think I rebased it but it didn't seem to have recorded it on my side - I preferred then to add manually the couple of changes missing. Hope the tests work now :)

mattjmeier reviewed (approved):
Thanks again for the contribution!!
